vc-got.el

```diff
@@ -593,5 +593,33 @@ Value is returned as floating point fractional number of days."
     (when (looking-at  "^[0-9]\\{1,\\}) \\([a-z0-9]+\\) \\([0-9]\\{4\\}-[0-9]\\{2\\}-[0-9]\\{2\\}\\) .+")
       (match-string-no-properties 1))))
 
+(defun vc-got-previous-revision (file rev)
+  "Return the revision number that precedes REV for FILE, or nil if no such revision exists."
+  (let (process-file-side-effects)
+    (vc-got-with-worktree file
+      (with-temp-buffer
+        (when (zerop (apply #'vc-got--call "log" (list "-c" rev "-l" "2" "-R" file)))
```

Comment:
I think you can use `vc-got--log` instead of applying `vc-got--call` and building a list by hand.  Sure, the code of `vc-got--log` can probably be simplified, but that's another issue :)

The advantage is also that vc-got--log will implicitly change the worktree, so you can also avoid the `vc-got-with-worktree`, and returns t or nil, so you can avoid also the zerop.  As an additional improvement we can move the `(let (process-file-side-effects) ...)` to `vc-got--log` to further reduce the boilerplate (from a quick check the log subcommand cannot alter the content of the source files in the worktree),

Reply:
I forgot to mention it in the commit message of 0a66694e897e35f0472d5ada455a76b7719cc285, but I added an argument for `vc-got--log` to require the log messages in reverse order (the "-R" flag) and to bind `process-file-sed-effects` :)

Reply:
I was trying following in vc-got-previous-revision:

```
(defun vc-got-previous-revision (file rev)
  "Return the revision number that precedes REV for FILE, or nil if no such revision exists."
  (with-temp-buffer
    (vc-got--log file 2 rev nil t)
    (keep-lines "^commit")
    (when (looking-at "^commit \\([a-z0-9]+\\)")
      (match-string-no-properties 1))))
```

But that gives error "Wrong type argument: stringp, t" and seems to point to rev argument to log for some reason.

Reply:
you forgot a nil before the last t.  the arguments now are `(path limit start-commit stop-commit search-patter reverse)`, so `(vc-got--log file 2 rev nil nil t)` should work.

Reply:
btw, the arguments for `vc-got--log` are perhaps too much, maybe it's time to switch to keyword arguments.